### PR TITLE
Class Decorators

### PR DESCRIPTION
**Pull request by inspecta**
- Created `Namable` class
- Inherited the `Namable` class in the `Person` class
- Created the base `Decorator` class
- Created both the `Capitalize` and `Trimmer` decorators

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
```
Comment (gemmen29) [inline capitalize_decorator.rb]: - [ ] Please be mindful that the `upcase` method capitalizes all of the passed string and not just the first letter as [required](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Implement%20a%20method%20correct_name%20that%20capitalizes%20the%20output%20of%20%40nameable.correct_name.).
### Hint: You can use the `capitalize` method.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @inspecta, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: capitalize_decorator.rb (hunk 1 above)
Comment (inspecta) [reply capitalize_decorator.rb]: Hi @gemmen29, thanks for the hint 😀

**Review by mckent05 — APPROVED** [review]
Hi @inspecta, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**